libpod/adapter/client.go

```diff
@@ -34,3 +34,14 @@ func (r RemoteRuntime) Connect() (*varlink.Connection, error) {
 	}
 	return connection, nil
 }
+
+// RefreshConnection is used to replace the current r.Conn after things like
+// using an upgraded varlink connection
+func (r RemoteRuntime) RefreshConnection() error {
```

Comment:
I would add a note in the comments detailing the connection state if r.Connect() fails?  Is the old connection still open? still upgraded? still ...